src/python/pants/vcs/git.py

```diff
@@ -124,42 +88,59 @@ def changed_files(
                 "--exclude-standard",
                 "--full-name",
             ] + rel_suffix
-            untracked = self._check_output(untracked_cmd)
+            untracked = self._git_binary.invoke(self._create_git_cmdline(untracked_cmd))
             files.update(untracked.split())
         # git will report changed files relative to the worktree: re-relativize to relative_to
         return {self._fix_git_relative_path(f, relative_to) for f in files}
 
     def changes_in(self, diffspec: str, relative_to: PurePath | str | None = None) -> set[str]:
         relative_to = PurePath(relative_to) if relative_to is not None else self.worktree
         cmd = ["diff-tree", "--no-commit-id", "--name-only", "-r", diffspec]
-        files = self._check_output(cmd).split()
+        files = self._git_binary.invoke(self._create_git_cmdline(cmd)).split()
         return {self._fix_git_relative_path(f.strip(), relative_to) for f in files}
 
-    # N.B.: Only used by tests.
-    def commit(self, message: str) -> None:
-        self._check_call(["commit", "--all", "--message", message])
+    def _create_git_cmdline(self, args: Iterable[str]) -> list[str]:
+        return [f"--git-dir={self._gitdir}", f"--work-tree={self.worktree}", *args]
 
-    # N.B.: Only used by tests.
-    def add(self, *paths: PurePath) -> None:
-        self._check_call(["add", *(str(path) for path in paths)])
 
-    def _check_call(self, args: Iterable[str]) -> None:
-        cmd = self._create_git_cmdline(args)
-        self._log_call(cmd)
-        result = subprocess.call(cmd)
-        self._check_result(cmd, result)
+@dataclass(frozen=True)
+class MaybeGitWorktree:
+    git_worktree: GitWorktree | None = None
```

Comment:
Both this type and the `GitWorktree` type should still extend `EngineAwareReturnType` with `def cacheable = False`. Because the `GitBinary` type will be re-computed once per-`pantsd` restart (i.e. "where is git on my `PATH`"), but the methods of `GitWorktree` need to be invalidated for every run of `pants` in case the `git` state has changed.